src/Features/CSharp/Portable/Completion/CompletionProviders/PropertySubPatternCompletionProvider.cs

```diff
@@ -0,0 +1,112 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Completion;
+using Microsoft.CodeAnalysis.Completion.Providers;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.CSharp.Completion.Providers
+{
+    internal class PropertySubpatternCompletionProvider : CommonCompletionProvider
+    {
+        public override async Task ProvideCompletionsAsync(CompletionContext context)
+        {
+            var document = context.Document;
+            var position = context.Position;
+            var cancellationToken = context.CancellationToken;
+            var tree = await document.GetSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
+
+            var token = TryGetOpenBraceOrCommaInPropertySubpattern(tree, position, cancellationToken);
+            if (token == default)
+            {
+                return;
+            }
+
+            var semanticModel = await document.GetSemanticModelForSpanAsync(new TextSpan(position, length: 0), cancellationToken).ConfigureAwait(false);
+            var pattern = (PatternSyntax)token.Parent.Parent;
+            var type = semanticModel.GetTypeInfo(pattern, cancellationToken).ConvertedType;
+            if (type == null)
+            {
+                return;
+            }
+
+            // Find the members that can be tested.
+            IEnumerable<ISymbol> members = semanticModel.LookupSymbols(position, type);
+            members = members.Where(m => m.CanBeReferencedByName &&
+                IsFieldOrReadableProperty(m) &&
+                !m.IsImplicitlyDeclared);
```

Comment:
can implicitly declared members ever be referenced by name?  (i'm guessing this is copied from elsewhere... if so, fine to leave as is).

Reply:
Yes, this is copied from completion logic for object initializers ;-)